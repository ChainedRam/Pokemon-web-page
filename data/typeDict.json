{
  "normal": {
    "name": "normal",
    "noDamageTo": ["ghost"],
    "halfDamageTo": ["rock", "steel"],
    "doubleDamageTo": [],
    "noDamageFrom": ["ghost"],
    "halfDamageFrom": [],
    "doubleDamageFrom": ["fighting"]
  },
  "fighting": {
    "name": "fighting",
    "noDamageTo": ["ghost"],
    "halfDamageTo": ["flying", "poison", "bug", "psychic", "fairy"],
    "doubleDamageTo": ["normal", "rock", "steel", "ice", "dark"],
    "noDamageFrom": [],
    "halfDamageFrom": ["rock", "bug", "dark"],
    "doubleDamageFrom": ["flying", "psychic", "fairy"]
  },
  "flying": {
    "name": "flying",
    "noDamageTo": [],
    "halfDamageTo": ["rock", "steel", "electric"],
    "doubleDamageTo": ["fighting", "bug", "grass"],
    "noDamageFrom": ["ground"],
    "halfDamageFrom": ["fighting", "bug", "grass"],
    "doubleDamageFrom": ["rock", "electric", "ice"]
  },
  "poison": {
    "name": "poison",
    "noDamageTo": ["steel"],
    "halfDamageTo": ["poison", "ground", "rock", "ghost"],
    "doubleDamageTo": ["grass", "fairy"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "poison", "bug", "grass", "fairy"],
    "doubleDamageFrom": ["ground", "psychic"]
  },
  "ground": {
    "name": "ground",
    "noDamageTo": ["flying"],
    "halfDamageTo": ["bug", "grass"],
    "doubleDamageTo": ["poison", "rock", "steel", "fire", "electric"],
    "noDamageFrom": ["electric"],
    "halfDamageFrom": ["poison", "rock"],
    "doubleDamageFrom": ["water", "grass", "ice"]
  },
  "rock": {
    "name": "rock",
    "noDamageTo": [],
    "halfDamageTo": ["fighting", "ground", "steel"],
    "doubleDamageTo": ["flying", "bug", "fire", "ice"],
    "noDamageFrom": [],
    "halfDamageFrom": ["normal", "flying", "poison", "fire"],
    "doubleDamageFrom": ["fighting", "ground", "steel", "water", "grass"]
  },
  "bug": {
    "name": "bug",
    "noDamageTo": [],
    "halfDamageTo": [
      "fighting",
      "flying",
      "poison",
      "ghost",
      "steel",
      "fire",
      "fairy"
    ],
    "doubleDamageTo": ["grass", "psychic", "dark"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "ground", "grass"],
    "doubleDamageFrom": ["flying", "rock", "fire"]
  },
  "ghost": {
    "name": "ghost",
    "noDamageTo": ["normal"],
    "halfDamageTo": ["dark"],
    "doubleDamageTo": ["ghost", "psychic"],
    "noDamageFrom": ["normal", "fighting"],
    "halfDamageFrom": ["poison", "bug"],
    "doubleDamageFrom": ["ghost", "dark"]
  },
  "steel": {
    "name": "steel",
    "noDamageTo": [],
    "halfDamageTo": ["steel", "fire", "water", "electric"],
    "doubleDamageTo": ["rock", "ice", "fairy"],
    "noDamageFrom": ["poison"],
    "halfDamageFrom": [
      "normal",
      "flying",
      "rock",
      "bug",
      "steel",
      "grass",
      "psychic",
      "ice",
      "dragon",
      "fairy"
    ],
    "doubleDamageFrom": ["fighting", "ground", "fire"]
  },
  "fire": {
    "name": "fire",
    "noDamageTo": [],
    "halfDamageTo": ["rock", "fire", "water", "dragon"],
    "doubleDamageTo": ["bug", "steel", "grass", "ice"],
    "noDamageFrom": [],
    "halfDamageFrom": ["bug", "steel", "fire", "grass", "ice", "fairy"],
    "doubleDamageFrom": ["ground", "rock", "water"]
  },
  "water": {
    "name": "water",
    "noDamageTo": [],
    "halfDamageTo": ["water", "grass", "dragon"],
    "doubleDamageTo": ["ground", "rock", "fire"],
    "noDamageFrom": [],
    "halfDamageFrom": ["steel", "fire", "water", "ice"],
    "doubleDamageFrom": ["grass", "electric"]
  },
  "grass": {
    "name": "grass",
    "noDamageTo": [],
    "halfDamageTo": [
      "flying",
      "poison",
      "bug",
      "steel",
      "fire",
      "grass",
      "dragon"
    ],
    "doubleDamageTo": ["ground", "rock", "water"],
    "noDamageFrom": [],
    "halfDamageFrom": ["ground", "water", "grass", "electric"],
    "doubleDamageFrom": ["flying", "poison", "bug", "fire", "ice"]
  },
  "electric": {
    "name": "electric",
    "noDamageTo": ["ground"],
    "halfDamageTo": ["grass", "electric", "dragon"],
    "doubleDamageTo": ["flying", "water"],
    "noDamageFrom": [],
    "halfDamageFrom": ["flying", "steel", "electric"],
    "doubleDamageFrom": ["ground"]
  },
  "psychic": {
    "name": "psychic",
    "noDamageTo": ["dark"],
    "halfDamageTo": ["steel", "psychic"],
    "doubleDamageTo": ["fighting", "poison"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "psychic"],
    "doubleDamageFrom": ["bug", "ghost", "dark"]
  },
  "ice": {
    "name": "ice",
    "noDamageTo": [],
    "halfDamageTo": ["steel", "fire", "water", "ice"],
    "doubleDamageTo": ["flying", "ground", "grass", "dragon"],
    "noDamageFrom": [],
    "halfDamageFrom": ["ice"],
    "doubleDamageFrom": ["fighting", "rock", "steel", "fire"]
  },
  "dragon": {
    "name": "dragon",
    "noDamageTo": ["fairy"],
    "halfDamageTo": ["steel"],
    "doubleDamageTo": ["dragon"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fire", "water", "grass", "electric"],
    "doubleDamageFrom": ["ice", "dragon", "fairy"]
  },
  "dark": {
    "name": "dark",
    "noDamageTo": [],
    "halfDamageTo": ["fighting", "dark", "fairy"],
    "doubleDamageTo": ["ghost", "psychic"],
    "noDamageFrom": ["psychic"],
    "halfDamageFrom": ["ghost", "dark"],
    "doubleDamageFrom": ["fighting", "bug", "fairy"]
  },
  "fairy": {
    "name": "fairy",
    "noDamageTo": [],
    "halfDamageTo": ["poison", "steel", "fire"],
    "doubleDamageTo": ["fighting", "dragon", "dark"],
    "noDamageFrom": ["dragon"],
    "halfDamageFrom": ["fighting", "bug", "dark"],
    "doubleDamageFrom": ["poison", "steel"]
  }
}
