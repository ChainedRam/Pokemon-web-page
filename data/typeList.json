[
  {
    "name": "normal",
    "weakness": ["fighting"]
  },
  {
    "name": "fighting",
    "weakness": ["flying", "psychic", "fairy"]
  },
  {
    "name": "flying",
    "weakness": ["rock", "electric", "ice"]
  },
  {
    "name": "poison",
    "weakness": ["ground", "psychic"]
  },
  {
    "name": "ground",
    "weakness": ["water", "grass", "ice"]
  },
  {
    "name": "rock",
    "weakness": ["fighting", "ground", "steel", "water", "grass"]
  },
  {
    "name": "bug",
    "weakness": ["flying", "rock", "fire"]
  },
  {
    "name": "ghost",
    "weakness": ["ghost", "dark"]
  },
  {
    "name": "steel",
    "weakness": ["fighting", "ground", "fire"]
  },
  {
    "name": "fire",
    "weakness": ["ground", "rock", "water"]
  },
  {
    "name": "water",
    "weakness": ["grass", "electric"]
  },
  {
    "name": "grass",
    "weakness": ["flying", "poison", "bug", "fire", "ice"]
  },
  {
    "name": "electric",
    "weakness": ["ground"]
  },
  {
    "name": "psychic",
    "weakness": ["bug", "ghost", "dark"]
  },
  {
    "name": "ice",
    "weakness": ["fighting", "rock", "steel", "fire"]
  },
  {
    "name": "dragon",
    "weakness": ["ice", "dragon", "fairy"]
  },
  {
    "name": "dark",
    "weakness": ["fighting", "bug", "fairy"]
  },
  {
    "name": "fairy",
    "weakness": ["poison", "steel"]
  }
]
