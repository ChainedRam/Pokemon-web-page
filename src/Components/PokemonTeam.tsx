import * as React from "react";

interface ITeamListProps {
  list: string[];
  onPokemonSelected(index: number, name: string): void;
}

export default class PokeTeam extends React.Component<ITeamListProps, {}> {
  public onSelect = (i, e) => {
    this.props.onPokemonSelected(i, e.currentTarget.value);
  };

  public render() {
    const maxTeam: any = [];
    const pokeItems = this.props.list.map((pokes, i) => (
      <option key={i}>{pokes}</option>
    ));
    for (let i = 0; i < 6; i++) {
      maxTeam.push(
        <li key={i}>
          <select
            className="btn btn-secondary"
            onChange={e => this.onSelect(i, e)}
          >
            {pokeItems}
          </select>
        </li>
      );
    }
    return (
      <div>
        <ul>{maxTeam}</ul>
      </div>
    );
  }
}
