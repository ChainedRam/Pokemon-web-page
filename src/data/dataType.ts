export interface Pokemon {
  name: string;
  types: Type[];
  learnableMoves: string[];
}

export interface PokemonDictionary {
  [pokeName: string]: Pokemon;
}

export interface Type {
  name: string;
  color: string;

  noDamageTo?: string[];
  halfDamageTo?: string[];
  doubleDamageTo?: string[];

  noDamageFrom?: string[];
  halfDamageFrom?: string[];
  doubleDamageFrom?: string[];
}

export interface TypeDictionary {
  [pokeName: string]: Type;
}

export function GetTypesDictionary(): TypeDictionary {
  return require("./typeDict.json");
}

export function GetPokeDictionary(): PokemonDictionary {
  let rawPokeDict = require("./pokeDict.json");
  let pokeDict: PokemonDictionary = {};
  let typeDict = GetTypesDictionary();
  for (let key in rawPokeDict) {
    pokeDict[key] = {
      name: key,
      types: rawPokeDict[key].types.map(t => typeDict[t]),
      learnableMoves: []
    };
  }

  return pokeDict;
}

export function GetPokemonList(): Pokemon[] {
  let pokeList: Pokemon[] = [];
  let typeDict = GetTypesDictionary();

  let rawPokeList = require("./pokeList.json") as any[];

  rawPokeList.forEach(p => {
    pokeList.push({
      name: p.name,
      types: p.types.map(t => typeDict[t]),
      learnableMoves: []
    });
  });

  return pokeList;
}
