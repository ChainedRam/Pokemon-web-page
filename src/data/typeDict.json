{
  "normal": {
    "name": "normal",
    "color": "A8A77A",
    "noDamageTo": ["ghost"],
    "halfDamageTo": ["rock", "steel"],
    "doubleDamageTo": [],
    "noDamageFrom": ["ghost"],
    "halfDamageFrom": [],
    "doubleDamageFrom": ["fighting"]
  },
  "fighting": {
    "name": "fighting",
    "color": "C22E28",
    "noDamageTo": ["ghost"],
    "halfDamageTo": ["flying", "poison", "bug", "psychic", "fairy"],
    "doubleDamageTo": ["normal", "rock", "steel", "ice", "dark"],
    "noDamageFrom": [],
    "halfDamageFrom": ["rock", "bug", "dark"],
    "doubleDamageFrom": ["flying", "psychic", "fairy"]
  },
  "flying": {
    "name": "flying",
    "color": "A98FF3",
    "noDamageTo": [],
    "halfDamageTo": ["rock", "steel", "electric"],
    "doubleDamageTo": ["fighting", "bug", "grass"],
    "noDamageFrom": ["ground"],
    "halfDamageFrom": ["fighting", "bug", "grass"],
    "doubleDamageFrom": ["rock", "electric", "ice"]
  },
  "poison": {
    "name": "poison",
    "color": "A33EA1",
    "noDamageTo": ["steel"],
    "halfDamageTo": ["poison", "ground", "rock", "ghost"],
    "doubleDamageTo": ["grass", "fairy"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "poison", "bug", "grass", "fairy"],
    "doubleDamageFrom": ["ground", "psychic"]
  },
  "ground": {
    "name": "ground",
    "color": "E2BF65",
    "noDamageTo": ["flying"],
    "halfDamageTo": ["bug", "grass"],
    "doubleDamageTo": ["poison", "rock", "steel", "fire", "electric"],
    "noDamageFrom": ["electric"],
    "halfDamageFrom": ["poison", "rock"],
    "doubleDamageFrom": ["water", "grass", "ice"]
  },
  "rock": {
    "name": "rock",
    "color": "B6A136",
    "noDamageTo": [],
    "halfDamageTo": ["fighting", "ground", "steel"],
    "doubleDamageTo": ["flying", "bug", "fire", "ice"],
    "noDamageFrom": [],
    "halfDamageFrom": ["normal", "flying", "poison", "fire"],
    "doubleDamageFrom": ["fighting", "ground", "steel", "water", "grass"]
  },
  "bug": {
    "name": "bug",
    "color": "A6B91A",
    "noDamageTo": [],
    "halfDamageTo": [
      "fighting",
      "flying",
      "poison",
      "ghost",
      "steel",
      "fire",
      "fairy"
    ],
    "doubleDamageTo": ["grass", "psychic", "dark"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "ground", "grass"],
    "doubleDamageFrom": ["flying", "rock", "fire"]
  },
  "ghost": {
    "name": "ghost",
    "color": "735797",
    "noDamageTo": ["normal"],
    "halfDamageTo": ["dark"],
    "doubleDamageTo": ["ghost", "psychic"],
    "noDamageFrom": ["normal", "fighting"],
    "halfDamageFrom": ["poison", "bug"],
    "doubleDamageFrom": ["ghost", "dark"]
  },
  "steel": {
    "name": "steel",
    "color": "B7B7CE",
    "noDamageTo": [],
    "halfDamageTo": ["steel", "fire", "water", "electric"],
    "doubleDamageTo": ["rock", "ice", "fairy"],
    "noDamageFrom": ["poison"],
    "halfDamageFrom": [
      "normal",
      "flying",
      "rock",
      "bug",
      "steel",
      "grass",
      "psychic",
      "ice",
      "dragon",
      "fairy"
    ],
    "doubleDamageFrom": ["fighting", "ground", "fire"]
  },
  "fire": {
    "name": "fire",
    "color": "EE8130",
    "noDamageTo": [],
    "halfDamageTo": ["rock", "fire", "water", "dragon"],
    "doubleDamageTo": ["bug", "steel", "grass", "ice"],
    "noDamageFrom": [],
    "halfDamageFrom": ["bug", "steel", "fire", "grass", "ice", "fairy"],
    "doubleDamageFrom": ["ground", "rock", "water"]
  },
  "water": {
    "name": "water",
    "color": "6390F0",
    "noDamageTo": [],
    "halfDamageTo": ["water", "grass", "dragon"],
    "doubleDamageTo": ["ground", "rock", "fire"],
    "noDamageFrom": [],
    "halfDamageFrom": ["steel", "fire", "water", "ice"],
    "doubleDamageFrom": ["grass", "electric"]
  },
  "grass": {
    "name": "grass",
    "color": "7AC74C",
    "noDamageTo": [],
    "halfDamageTo": [
      "flying",
      "poison",
      "bug",
      "steel",
      "fire",
      "grass",
      "dragon"
    ],
    "doubleDamageTo": ["ground", "rock", "water"],
    "noDamageFrom": [],
    "halfDamageFrom": ["ground", "water", "grass", "electric"],
    "doubleDamageFrom": ["flying", "poison", "bug", "fire", "ice"]
  },
  "electric": {
    "name": "electric",
    "color": "F7D02C",
    "noDamageTo": ["ground"],
    "halfDamageTo": ["grass", "electric", "dragon"],
    "doubleDamageTo": ["flying", "water"],
    "noDamageFrom": [],
    "halfDamageFrom": ["flying", "steel", "electric"],
    "doubleDamageFrom": ["ground"]
  },
  "psychic": {
    "name": "psychic",
    "color": "F95587",
    "noDamageTo": ["dark"],
    "halfDamageTo": ["steel", "psychic"],
    "doubleDamageTo": ["fighting", "poison"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "psychic"],
    "doubleDamageFrom": ["bug", "ghost", "dark"]
  },
  "ice": {
    "name": "ice",
    "color": "96D9D6",
    "noDamageTo": [],
    "halfDamageTo": ["steel", "fire", "water", "ice"],
    "doubleDamageTo": ["flying", "ground", "grass", "dragon"],
    "noDamageFrom": [],
    "halfDamageFrom": ["ice"],
    "doubleDamageFrom": ["fighting", "rock", "steel", "fire"]
  },
  "dragon": {
    "name": "dragon",
    "color": "6F35FC",
    "noDamageTo": ["fairy"],
    "halfDamageTo": ["steel"],
    "doubleDamageTo": ["dragon"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fire", "water", "grass", "electric"],
    "doubleDamageFrom": ["ice", "dragon", "fairy"]
  },
  "dark": {
    "name": "dark",
    "color": "705746",
    "noDamageTo": [],
    "halfDamageTo": ["fighting", "dark", "fairy"],
    "doubleDamageTo": ["ghost", "psychic"],
    "noDamageFrom": ["psychic"],
    "halfDamageFrom": ["ghost", "dark"],
    "doubleDamageFrom": ["fighting", "bug", "fairy"]
  },
  "fairy": {
    "name": "fairy",
    "color": "D685AD",
    "noDamageTo": [],
    "halfDamageTo": ["poison", "steel", "fire"],
    "doubleDamageTo": ["fighting", "dragon", "dark"],
    "noDamageFrom": ["dragon"],
    "halfDamageFrom": ["fighting", "bug", "dark"],
    "doubleDamageFrom": ["poison", "steel"]
  }
}
