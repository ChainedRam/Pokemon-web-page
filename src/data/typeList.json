[
  {
    "name": "normal",
    "color": "A8A77A",
    "noDamageTo": ["ghost"],
    "halfDamageTo": ["rock", "steel"],
    "doubleDamageTo": [],
    "noDamageFrom": ["ghost"],
    "halfDamageFrom": [],
    "doubleDamageFrom": ["fighting"]
  },
  {
    "name": "fighting",
    "color": "C22E28",
    "noDamageTo": ["ghost"],
    "halfDamageTo": ["flying", "poison", "bug", "psychic", "fairy"],
    "doubleDamageTo": ["normal", "rock", "steel", "ice", "dark"],
    "noDamageFrom": [],
    "halfDamageFrom": ["rock", "bug", "dark"],
    "doubleDamageFrom": ["flying", "psychic", "fairy"]
  },
  {
    "name": "flying",
    "color": "A98FF3",
    "noDamageTo": [],
    "halfDamageTo": ["rock", "steel", "electric"],
    "doubleDamageTo": ["fighting", "bug", "grass"],
    "noDamageFrom": ["ground"],
    "halfDamageFrom": ["fighting", "bug", "grass"],
    "doubleDamageFrom": ["rock", "electric", "ice"]
  },
  {
    "name": "poison",
    "color": "A33EA1",
    "noDamageTo": ["steel"],
    "halfDamageTo": ["poison", "ground", "rock", "ghost"],
    "doubleDamageTo": ["grass", "fairy"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "poison", "bug", "grass", "fairy"],
    "doubleDamageFrom": ["ground", "psychic"]
  },
  {
    "name": "ground",
    "color": "E2BF65",
    "noDamageTo": ["flying"],
    "halfDamageTo": ["bug", "grass"],
    "doubleDamageTo": ["poison", "rock", "steel", "fire", "electric"],
    "noDamageFrom": ["electric"],
    "halfDamageFrom": ["poison", "rock"],
    "doubleDamageFrom": ["water", "grass", "ice"]
  },
  {
    "name": "rock",
    "color": "B6A136",
    "noDamageTo": [],
    "halfDamageTo": ["fighting", "ground", "steel"],
    "doubleDamageTo": ["flying", "bug", "fire", "ice"],
    "noDamageFrom": [],
    "halfDamageFrom": ["normal", "flying", "poison", "fire"],
    "doubleDamageFrom": ["fighting", "ground", "steel", "water", "grass"]
  },
  {
    "name": "bug",
    "color": "A6B91A",
    "noDamageTo": [],
    "halfDamageTo": [
      "fighting",
      "flying",
      "poison",
      "ghost",
      "steel",
      "fire",
      "fairy"
    ],
    "doubleDamageTo": ["grass", "psychic", "dark"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "ground", "grass"],
    "doubleDamageFrom": ["flying", "rock", "fire"]
  },
  {
    "name": "ghost",
    "color": "735797",
    "noDamageTo": ["normal"],
    "halfDamageTo": ["dark"],
    "doubleDamageTo": ["ghost", "psychic"],
    "noDamageFrom": ["normal", "fighting"],
    "halfDamageFrom": ["poison", "bug"],
    "doubleDamageFrom": ["ghost", "dark"]
  },
  {
    "name": "steel",
    "color": "B7B7CE",
    "noDamageTo": [],
    "halfDamageTo": ["steel", "fire", "water", "electric"],
    "doubleDamageTo": ["rock", "ice", "fairy"],
    "noDamageFrom": ["poison"],
    "halfDamageFrom": [
      "normal",
      "flying",
      "rock",
      "bug",
      "steel",
      "grass",
      "psychic",
      "ice",
      "dragon",
      "fairy"
    ],
    "doubleDamageFrom": ["fighting", "ground", "fire"]
  },
  {
    "name": "fire",
    "color": "EE8130",
    "noDamageTo": [],
    "halfDamageTo": ["rock", "fire", "water", "dragon"],
    "doubleDamageTo": ["bug", "steel", "grass", "ice"],
    "noDamageFrom": [],
    "halfDamageFrom": ["bug", "steel", "fire", "grass", "ice", "fairy"],
    "doubleDamageFrom": ["ground", "rock", "water"]
  },
  {
    "name": "water",
    "color": "6390F0",
    "noDamageTo": [],
    "halfDamageTo": ["water", "grass", "dragon"],
    "doubleDamageTo": ["ground", "rock", "fire"],
    "noDamageFrom": [],
    "halfDamageFrom": ["steel", "fire", "water", "ice"],
    "doubleDamageFrom": ["grass", "electric"]
  },
  {
    "name": "grass",
    "color": "7AC74C",
    "noDamageTo": [],
    "halfDamageTo": [
      "flying",
      "poison",
      "bug",
      "steel",
      "fire",
      "grass",
      "dragon"
    ],
    "doubleDamageTo": ["ground", "rock", "water"],
    "noDamageFrom": [],
    "halfDamageFrom": ["ground", "water", "grass", "electric"],
    "doubleDamageFrom": ["flying", "poison", "bug", "fire", "ice"]
  },
  {
    "name": "electric",
    "color": "F7D02C",
    "noDamageTo": ["ground"],
    "halfDamageTo": ["grass", "electric", "dragon"],
    "doubleDamageTo": ["flying", "water"],
    "noDamageFrom": [],
    "halfDamageFrom": ["flying", "steel", "electric"],
    "doubleDamageFrom": ["ground"]
  },
  {
    "name": "psychic",
    "color": "F95587",
    "noDamageTo": ["dark"],
    "halfDamageTo": ["steel", "psychic"],
    "doubleDamageTo": ["fighting", "poison"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fighting", "psychic"],
    "doubleDamageFrom": ["bug", "ghost", "dark"]
  },
  {
    "name": "ice",
    "color": "96D9D6",
    "noDamageTo": [],
    "halfDamageTo": ["steel", "fire", "water", "ice"],
    "doubleDamageTo": ["flying", "ground", "grass", "dragon"],
    "noDamageFrom": [],
    "halfDamageFrom": ["ice"],
    "doubleDamageFrom": ["fighting", "rock", "steel", "fire"]
  },
  {
    "name": "dragon",
    "color": "6F35FC",
    "noDamageTo": ["fairy"],
    "halfDamageTo": ["steel"],
    "doubleDamageTo": ["dragon"],
    "noDamageFrom": [],
    "halfDamageFrom": ["fire", "water", "grass", "electric"],
    "doubleDamageFrom": ["ice", "dragon", "fairy"]
  },
  {
    "name": "dark",
    "color": "705746",
    "noDamageTo": [],
    "halfDamageTo": ["fighting", "dark", "fairy"],
    "doubleDamageTo": ["ghost", "psychic"],
    "noDamageFrom": ["psychic"],
    "halfDamageFrom": ["ghost", "dark"],
    "doubleDamageFrom": ["fighting", "bug", "fairy"]
  },
  {
    "name": "fairy",
    "color": "D685AD",
    "noDamageTo": [],
    "halfDamageTo": ["poison", "steel", "fire"],
    "doubleDamageTo": ["fighting", "dragon", "dark"],
    "noDamageFrom": ["dragon"],
    "halfDamageFrom": ["fighting", "bug", "dark"],
    "doubleDamageFrom": ["poison", "steel"]
  }
]
