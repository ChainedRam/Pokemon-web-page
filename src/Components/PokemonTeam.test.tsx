import * as React from "react";
import * as Renderer from "react-test-renderer";
import PokemonTeam from "./PokemonTeam";
import { Pokemon } from "../data/dataType";

describe("Pokemon Team component", () => {
  let pokemonList: Pokemon[] = [
    {
      name: "fakemon 1",
      learnSet: ["attack"],
      types: [{ name: "water", color: "blue" }],
      abilitySet: []
    },
    {
      name: "fakemon 2",
      learnSet: ["attack"],
      types: [{ name: "fire", color: "red" }],
      abilitySet: []
    },
    {
      name: "fakemon 3",
      learnSet: ["attack"],
      types: [{ name: "water", color: "blue" }, { name: "fire", color: "red" }],
      abilitySet: []
    }
  ];

  it("renders as snapshot", () => {
    let pokemonTeam = Renderer.create(
      <PokemonTeam
        pokemonTeam={[null]}
        selectablePokemon={pokemonList}
        onPokemonSelected={() => {}}
      />
    ).toJSON();

    expect(pokemonTeam).toMatchSnapshot();
  });
});
