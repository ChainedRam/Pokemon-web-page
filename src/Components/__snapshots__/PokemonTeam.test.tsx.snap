// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Team component renders as snapshot 1`] = `
<ul
  className="list-group"
>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Select pokemon
    </button>
    <div
      aria-hidden={true}
      className="dropdown-menu"
      role="menu"
      tabIndex="-1"
      x-placement={undefined}
    >
      <button
        className="dropdown-item"
        onClick={[Function]}
        tabIndex="0"
        type="button"
        value={0}
      >
        fakemon 1
      </button>
      <button
        className="dropdown-item"
        onClick={[Function]}
        tabIndex="0"
        type="button"
        value={1}
      >
        fakemon 2
      </button>
      <button
        className="dropdown-item"
        onClick={[Function]}
        tabIndex="0"
        type="button"
        value={2}
      >
        fakemon 3
      </button>
    </div>
  </div>
</ul>
`;
